src/lib/translators/mongo.ts

```diff
@@ -545,11 +548,22 @@ const mapper: Partial<StepMatcher<MongoStep>> = {
     },
   }),
   filter: filterstepToMatchstep,
-  formula: (step: Readonly<S.FormulaStep>) => ({
-    $addFields: {
-      [step.new_column]: buildMongoFormulaTree(math.parse(step.formula)),
-    },
-  }),
+  formula: (step: Readonly<S.FormulaStep>) => {
+    // replace all `[column name]` values by a pseudo and store it in `pseudos`
+    // `buildMongoFormulaTree` will re-put the name back
+    const pseudos: { [key: string]: string } = {};
+    const formulaPseudotised = step.formula.replace(/\[.*?\]/g, columName => {
+      const pseudo = `column_n_${Object.keys(pseudos).length}`;
+      pseudos[pseudo] = columName.replace('[', '').replace(']', '');
```

Comment:
I might be overlooking but it seems that using twice the same "column" name in a single formula would generate two distinct pseudos and I think it would be nice to always use the same. What do you think?

Besides, is everything translated correctly if there are several formula steps in a single pipeline? Are there any risk of overwriting a given "column"?